Repository/Api/EditableRepository.php

```diff
@@ -39,4 +39,13 @@
      * @throws UnsupportedLanguageException If the language is not supported.
      */
     public function move($sourceQuery, $targetPath, $language = 'glob');
+
+    /**
+     * Change the position of a node relative to its siblings.
+     *
+     * @param string $sourcePath
+     * @param int $position
+     */
+    public function reposition($sourcePath, $position);
```

Comment:
Was debating the form of this method - traditionally we specify the target `name` and move it before that. The ODM adds an adds a `$before` parameter, but its just a convenience, as under the hood it PHPCR has no such option.

In practical cases it is easier to specify the absolute position index instead of a name - as when specifying a name you always need to take into account if you are moving the node to the end of the list (or the beginning depending on how you implement it).

With an absolute position you have no such problems. There is a chance of concurrency issues (somebody moves the nodes while you make the request) but these apply equally to the name method - the only difference being that relative consistency is achieved instead of absolute. I don't see any advantage of one over the other ?